types_json.go

```diff
@@ -0,0 +1,104 @@
+package spec
+
```

Comment:
why do we need json representation? can this be automated via generate?

Reply:
why do you need json encoding in the first place?

Reply:
but why use json? for readability? 
Btw, you can just encode with the default golang encoding? 

(It means more code to maintain in the spec which is not ideal)